lib/io/hyped_spi/hyped_spi_derive/src/lib.rs

```diff
@@ -13,7 +13,68 @@ pub fn hyped_spi_derive(input: TokenStream) -> TokenStream {
 fn impl_hyped_spi(ast: &syn::DeriveInput) -> TokenStream {
     let name = &ast.ident;
     let generics: &Generics = &ast.generics;
-    let (_impl_generics, _ty_generics, _) = generics.split_for_impl();
-    let gen = quote! {};
+    let (impl_generics, ty_generics, _) = generics.split_for_impl();
+    let gen = quote! {
+        impl #impl_generics HypedSpi for #name #ty_generics{
+            fn read(&mut self, words: &mut [u8]) -> Result<(), SpiError> {
```

Comment:
is there any way not to have out params?
prob
```rs
fn ... -> Result<Vec<u8, 64>, SpiError>
```
or 
```rs
fn ... -> Result<(Vec<u8, 64>, usize), SpiError>
```
where `usize` is the number of set bytes?

i assume you tried something like this though